### DISCUSSION TITLE
How to Show All User

### DISCUSSION THREAD

**Discussion by Junkwarior23** [codeigniter4/shield#956] [Q&A] [answered]
I just try codeignniter shield and i want my admin can see all avaiable user, but i dont know what to do. can any body help me

**Comment by kenjis** [comment]
Shield has a command to list users:
https://codeigniter4.github.io/shield/user_management/managing_users/#managing-users-via-cli

You can see the code:
https://github.com/codeigniter4/shield/blob/fb4142bd32ea42059062be8508cc2925c52c0c93/src/Commands/User.php#L515

**Reply by Junkwarior23** [reply]
i'm sorry but is there any way to using code? i need it for my superadmin can see all user and also change user to admin via browser

**Reply by datamweb** [reply]
Shield does not provide **Admin** area by default. You need to create a custom controller. And use Shield functions.

**Reply by Junkwarior23** [reply]
i think there is no function to show all user with their group, you can show current user but not other

**Reply by datamweb** [reply, marked as answer]
> show all user with their group

This is a simple example, of course what you need is a little more knowledge. You may need to extend the [model](https://codeigniter4.github.io/shield/customization/user_provider/) or even `CodeIgniter\Shield\Entities\User`. Or use the table [join](https://codeigniter.com/user_guide/database/query_builder.html?highlight=join#id16).
```php
$users = auth()->getProvider();

// table 
$table = new \CodeIgniter\View\Table();
$table->setHeading(['username', 'Groups']);

foreach ($users->findAll() as $user) {
$groups =  implode(",", $user->getGroups());
$table->addRow([$user->username,$groups]);
echo $table->generate();
}
```

**Reply by Junkwarior23** [reply]
Thanks for the reply it work 
Can I ask something? I need to add name to my users, but I forgot how to do it. I can do it in my last project but I forgot everything I need to do it

**Reply by datamweb** [reply]
> Can I ask something?

Please ask each question in a separate discussions, we expect users who come to this section to be able to search for the solution with the titles.

>  I need to add name to my users

See https://github.com/codeigniter4/shield/discussions/333#discussioncomment-3311090
